File: 1283-find-the-smallest-divisor-given-a-threshold/1283-find-the-smallest-divisor-given-a-threshold.py
class Solution:
    def smallestDivisor(self, nums: List[int], threshold: int) -> int:
        l, r = 1, max(nums)
        
        while l<=r:
            mid = (l+r) //2
            total_sum = 0 
            for n in nums:
                total_sum += math.ceil(n/mid)
            if total_sum > threshold:
                l = mid + 1
            else:
                r = mid -1
        return l
                
                
        

File: 1283-find-the-smallest-divisor-given-a-threshold/1283-find-the-smallest-divisor-given-a-threshold.js
/**
 * @param {number[]} nums
 * @param {number} threshold
 * @return {number}
 */
var smallestDivisor = function(nums, threshold) {
    let l = 1
    let r = Math.max(...nums)
    
    while (l<=r){
        let mid = Math.floor((l+r)/2);
        let sum1 = 0
        for (let n of nums){
            sum1 += Math.ceil(n/mid);
        }
        if (sum1 > threshold){
            l =  mid + 1
        }else{
            r = mid -1
        }
        
    }
    return l
};